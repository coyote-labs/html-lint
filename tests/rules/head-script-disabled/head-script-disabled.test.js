const {htmlLint} = require('../../../index');

global.console = {
  error: jest.fn()
};

test.each([
  'tests/rules/head-script-disabled/error.html'
])(
  'head-script-disabled %s',
  async(fixture) => {
    try {
      await htmlLint(fixture);
    } catch ({ message }) {
      expect(global.console.error).toMatchSnapshot();
    }
  },
);
