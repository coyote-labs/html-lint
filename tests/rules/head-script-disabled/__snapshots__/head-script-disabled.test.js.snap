// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`head-script-disabled tests/rules/head-script-disabled/error.html 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      "[101m[html-lint] 3 errors in tests/rules/head-script-disabled/error.html[49m",
    ],
    Array [
      "[90mdoctype-first: HTML files should begin with a \`doctype\`.[39m
[0m[31m[1m>[22m[39m[90m 1 | [39m[33m<[39m[33mhtml[39m[33m>[39m[0m
[0m [90m   | [39m[31m[1m^[22m[39m [31m[1mHTML files should begin with a \`doctype\`.[22m[39m[0m
[0m [90m 2 | [39m  [33m<[39m[33mhead[39m[33m>[39m[0m
[0m [90m 3 | [39m    [33m<[39m[33mtitle[39m[33m>[39m[33mTest[39m[33m<[39m[33m/[39m[33mtitle[39m[33m>[39m[0m
[0m [90m 4 | [39m    [33m<[39m[33mscript[39m src[33m=[39m[32m\\"sample.js\\"[39m[33m>[39m[33m<[39m[33m/[39m[33mscript[39m[33m>[39m[0m",
      "
",
    ],
    Array [
      "[90mhead-script-disabled: Avoid using <script> tags in <head> without the \`defer\` or \`async\` attributes.[39m
[0m [90m 2 | [39m  [33m<[39m[33mhead[39m[33m>[39m[0m
[0m [90m 3 | [39m    [33m<[39m[33mtitle[39m[33m>[39m[33mTest[39m[33m<[39m[33m/[39m[33mtitle[39m[33m>[39m[0m
[0m[31m[1m>[22m[39m[90m 4 | [39m    [33m<[39m[33mscript[39m src[33m=[39m[32m\\"sample.js\\"[39m[33m>[39m[33m<[39m[33m/[39m[33mscript[39m[33m>[39m[0m
[0m [90m   | [39m    [31m[1m^[22m[39m [31m[1mAvoid using <script> tags in <head> without the \`defer\` or \`async\` attributes.[22m[39m[0m
[0m [90m 5 | [39m    [33m<[39m[33mscript[39m src[33m=[39m[32m\\"sample.js\\"[39m async[33m>[39m[33m<[39m[33m/[39m[33mscript[39m[33m>[39m[0m
[0m [90m 6 | [39m    [33m<[39m[33mscript[39m src[33m=[39m[32m\\"sample.js\\"[39m defer[33m>[39m[33m<[39m[33m/[39m[33mscript[39m[33m>[39m[0m
[0m [90m 7 | [39m    [33m<[39m[33mscript[39m[33m>[39mconsole[33m.[39mlog([32m'hello'[39m)[33m<[39m[33m/[39m[33mscript[39m[33m>[39m[0m",
      "
",
    ],
    Array [
      "[90mhead-script-disabled: Avoid using <script> tags in <head> without the \`defer\` or \`async\` attributes.[39m
[0m [90m  5 | [39m    [33m<[39m[33mscript[39m src[33m=[39m[32m\\"sample.js\\"[39m async[33m>[39m[33m<[39m[33m/[39m[33mscript[39m[33m>[39m[0m
[0m [90m  6 | [39m    [33m<[39m[33mscript[39m src[33m=[39m[32m\\"sample.js\\"[39m defer[33m>[39m[33m<[39m[33m/[39m[33mscript[39m[33m>[39m[0m
[0m[31m[1m>[22m[39m[90m  7 | [39m    [33m<[39m[33mscript[39m[33m>[39mconsole[33m.[39mlog([32m'hello'[39m)[33m<[39m[33m/[39m[33mscript[39m[33m>[39m[0m
[0m [90m    | [39m    [31m[1m^[22m[39m [31m[1mAvoid using <script> tags in <head> without the \`defer\` or \`async\` attributes.[22m[39m[0m
[0m [90m  8 | [39m    [33m<[39m[33mscript[39m async[33m>[39mconsole[33m.[39mlog([32m'hello'[39m)[33m<[39m[33m/[39m[33mscript[39m[33m>[39m[0m
[0m [90m  9 | [39m    [33m<[39m[33mscript[39m defer[33m>[39mconsole[33m.[39mlog([32m'hello'[39m)[33m<[39m[33m/[39m[33mscript[39m[33m>[39m[0m
[0m [90m 10 | [39m  [33m<[39m[33m/[39m[33mhead[39m[33m>[39m[0m",
      "
",
    ],
    Array [
      "
",
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
    Object {
      "type": "return",
      "value": undefined,
    },
    Object {
      "type": "return",
      "value": undefined,
    },
    Object {
      "type": "return",
      "value": undefined,
    },
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;
